test/integration.sh

```diff
@@ -1,47 +1,93 @@
+       command eval  'containers_expected=($(cat $fname))'
+    containers_expected=($containers_expected)  # cast as array
 
-    [[ -d outputs ]] && sudo rm -rf outputs && INFO "deleted old outputs"
+    num_containers_exp="${#containers_expected[@]}"
 
-    access_token=$(../redwood/cli/bin/redwood token create)
-    INFO "generated testing access token: ${access_token}"
+    # Crude initial test to verify that at least the minimum number of
+    # containers are running.
+    [ "${#containers[@]}" -ge $num_containers_exp ] ||\
+        (ERROR "Not running the required number of docker containers."\
+		&& exit 1)
 
-    redwood_endpoint=$(cat ../redwood/.env | grep 'base_url=' | sed 's/[^=]*=//')
-    INFO "got redwood endpoint from dcc-ops/redwood/.env: ${redwood_endpoint}"
+    for container_expected in "${containers_expected[@]}"; do
+        counter=0
+        for container in "${containers[@]}"; do
+            if [[ $container_expected = $container ]]; then
+                continue
+            else
+                counter=$((counter+1))
+            fi
+        done
 
-    INFO "doing upload with $(pwd)/manifest.tsv"
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/manifest.tsv:/dcc/manifest.tsv -v $(pwd)/samples:/samples -v $(pwd)/outputs:/outputs \
-         quay.io/ucsc_cgl/core-client:1.1.0-alpha spinnaker-upload --skip-submit --force-upload /dcc/manifest.tsv
+        if [[ "$counter" -eq $num_containers_exp ]]; then
+            ERROR "One or more docker containers not running."\
+		&& exit 1
+        fi
+    done
+
+}
+
+function main {
+    # This tests whether all Docker containers are installed and then
+    # hits the DCC Dashboard Host and checks whether it returns a 200
+    # status code.
+
+    cd "$(dirname "${BASH_SOURCE[0]}")"
+    LOG_LEVEL_ALL
+    check_setup
 
-    object_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 20)
-    bundle_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 19)
-    INFO "the uploaded bundle with id ${bundle_id} has metadata.json with object id ${object_id}"
+    # Get name of the DCC Dashboard Host.
+    curr_dir=$(pwd)
+    cd ../boardwalk
+    dcc_host=$(grep -o 'dcc_dashboard_service.*' .env | cut -f2- -d=)
+    # Check from environment variable value whether user is white-listed.    
+    whitelist_set=$(grep -o 'email_white.*' .env | cut -f2- -d=)
+    cd "$curr_dir"
 
-    INFO "downloading metadata.json with object id ${object_id}"
-    sudo mkdir outputs/test_download
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/outputs:/outputs quay.io/ucsc_cgl/core-client:1.1.0-alpha download ${object_id} /outputs/test_download
+    # Check if home page is returned. If so it will not output anything,
+    # otherwise it aborts and prints cURL error code to stdout.
+    curl -fsS "https://$dcc_host" > /dev/null
```

Comment:
Can you just get rid of this line? You are executing the curl twice now, once here, and once on line 72.

Reply:
Actually, that's intended. The `curl` in line 69 will abort the script if it can't resolve the URL (`curl` error code 6), i.e., the server doesn't exist. If the server does exist the script keeps executing. The `curl` in line 72 then tests the URL and exits the script and returns HTTP status codes if it isn't 200. 

I can do more research, but this is the best I could come up with for now. What do you think?